.github/workflows/python-package.yml

```diff
@@ -14,50 +18,24 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.9', '3.10', '3.11']
     if: "!contains(github.event.head_commit.message, '[ci skip]')"
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
-    - uses: Gr1N/setup-poetry@v8
-      with:
-        poetry-version: 1.1.13
-    - name: Install dependencies
+    - name: Install dependencies and build
       run: |
-        poetry build
-        pip install pytest pytest-cov codecov
-        pip install dist/*.whl
+        pip install -e '.[dev]'
+        python -m build
     - name: Test with pytest
       run: |
-        pytest
+        cd ./dist
+        pytest ../
     - uses: actions/upload-artifact@v2
       with:
```

Comment:
upload-artifact is up to ver.4 now but requires `overwrite: true` to work as before

Reply:
changed